### PR TITLE
Add snapshot featuregate support

### PR DESCRIPTION
**Pull request by awels**
Signed-off-by: Alexander Wels <awels@redhat.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Enhancement: Snapshotting is now a feature gate.
```

### CODE REVIEW

File: pkg/controller/hostpathprovisioner/controller.go [hunk 1]
```diff
@@ -44,14 +44,21 @@ import (
...
+const (
+	snapshotFeatureGate = "Snapshotting"
+)
 
 func isErrCacheNotStarted(err error) bool {
 	if err == nil {
 		return false
 	}
-	return err.(*cache.ErrCacheNotStarted) != nil
+	_, ok := err.(*cache.ErrCacheNotStarted)
```
Comment (akalenyu) [inline pkg/controller/hostpathprovisioner/controller.go:60]: If we go this way, shouldn't we remove the asterisk?
https://github.com/docker/docker-credential-helpers/blob/master/credentials/error.go#L32-L35

**Comment by akalenyu** [comment]
Other than the only question I posted lgtm

File: pkg/controller/hostpathprovisioner/controller.go (hunk 1 above)
Comment (awels) [reply pkg/controller/hostpathprovisioner/controller.go:60]: No the 'ok' is only true if cache.ErrCacheNotStarted  is not nil, and *cache.ErrCacheNotStarted is true. It has the npe protection build in.

Comment (awels) [reply pkg/controller/hostpathprovisioner/controller.go:60]: Its part of the go type assertion logic.

**Pull request merged**